### DISCUSSION TITLE
Questions regarding the Support of RIS

### DISCUSSION THREAD

**Discussion by ekilcioglu** [NVlabs/sionna#488] [Q&A] [answered]
Hello,

First of all, thank you so much for the addition of RIS into the ray-tracing tool. After reviewing the documentation of the RIS, I have three questions regarding this.

1) In our project, we realize RIS as a continuous surface. Previously, before the addition of RIS support, for the analysis in the ray-tracing tool, I realized RIS as a discretized version with each unit cell equal to 0.1 lambda. While introducing RIS (as a receiver or as a transmitter for TX-RIS and RIS-RX channels separately), I used 0.1 lambda spacing for vertical_spacing and horizontal_spacing. However, with the new version that includes RIS, I understand that RIS is introduced with a default spacing of 0.5 lambda. My question is, can we somehow change this spacing to a different scale of lambda (for example, 0.1 lambda)? This is necessary because in the practical scenario, we will have almost no space between cells as it will be treated as a continuous surface, and I want the simulation to closely match our practical scenario.

2) You have implemented two great approaches from the literature to decide on the phase of each unit cell. Can we directly apply our phase assigning method by directly setting "ris.phase_profile" and use these assigned phase values for the coverage map analysis?

3) Before the support of RIS, as I mentioned, I implemented RIS-assisted scenarios in two different channels (TX-RIS and RIS-RX). For the RIS-RX channel, I calculated the RIS's coverage map using the following code, by defining precoding_vec as unity power instead of dividing the unit power among all unit cells like a classical antenna array does:

cm = scene.coverage_map(cm_cell_size=[0.1,0.1], num_samples=int(1e7), max_depth=5, precoding_vec=[1 for _ in range(num_rows_RIS * num_cols_RIS)])

Is this effect already considered in the RIS support while being calculated in the coverage map including the RIS?

Thank you for your assistance.

Best regards

**Comment by jhoydis** [comment, marked as answer]
Hi @ekilcioglu,

1) No, the spacing of unit cells cannot be changed. This is based on the conditions mentioned in 
Vittorio Degli-Esposti et al., “[Reradiation and Scattering From a Reconfigurable Intelligent Surface: A General Macroscopic Model](https://ieeexplore.ieee.org/abstract/document/9713744)”, IEEE Trans. Antennas Propag, vol. 70, no. 10, pp.8691-8706, Oct. 2022.

You can of course modify the source code to change the spacing if this is what you need. Simply look at the ris.py file and change the code in the spacing property of all classes.

2) Yes. You can assign directly values to the phase and amplitude profiles. The helper functions are just there as examples from the literature.

3) I am not sure to understand this question. The coverage maps are calculated by treating the RIS as an anomalous reflector. There is no notion of precoding vector.

**Comment by ekilcioglu** [comment]
Hello Mr. Hoydis,

I appreciate your quick reply. I will review the "ris.py" file and try to change the spacing manually and see how it goes. Thanks! For the third question, I understand the idea right now, so it is my misunderstanding. Thank you once again for your answers,

Best regards,
Emre
